Src/VimCore/CommandUtil.fs

```diff
@@ -1970,6 +1970,18 @@ type internal CommandUtil
                     col |> Seq.iter (fun span -> edit.Delete(span) |> ignore)
                     edit.Apply() |> ignore
 
+                    // For character-wise put of simple text over a
+                    // block selection, replicate the text into a block.
+                    let newStringData =
+                        match operationKind, stringData with
+                        | OperationKind.CharacterWise, StringData.Simple text ->
+                            seq { for _ in col do yield text }
+                            |> NonEmptyCollectionUtil.OfSeq 
+                            |> Option.get

```

Comment:
Hmm. May be able to simplify these first three lines to:

``` fsharp
NonEmptyCollectionUtil.Map (fun _ -> text) col 
```

That would avoid the `Option.get`.